### PR TITLE
Test that the AI will select a Play for every RefboxGameState

### LINKED ISSUE #476: Add Unit test to make sure we have a Play for every refbox and world state

### PR DESCRIPTION
**Pull request by jordanzliu**
### Description

Test that the AI will select a Play for every RefboxGameState

### Testing Done

Ran the test

### Resolved Issues

Resolves #476 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/test/ai/game_state_play_selection.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        Robot robot_2(2, Point(0, 5.0), Point(), Angle::zero(), AngularVelocity::zero(),
+                      Timestamp::fromSeconds(0));
+        world.mutableFriendlyTeam().updateRobots({robot_0, robot_1, robot_2});
+
+        Robot enemy_robot_0(0, Point(-1.1, 1), Point(), Angle::zero(), AngularVelocity::zero(),
+                      Timestamp::fromSeconds(0));
+        Robot enemy_robot_1(1, Point(2, 0.81), Point(), Angle::zero(), AngularVelocity::zero(),
+                      Timestamp::fromSeconds(0));
+        Robot enemy_robot_2(2, Point(0, 5.0), Point(), Angle::zero(), AngularVelocity::zero(),
+                      Timestamp::fromSeconds(0));
```
Comment (garethellis0) [inline src/thunderbots/software/test/ai/game_state_play_selection.cpp]: Maybe we should invert the `x` values for friendlies so the robots aren't on top of each other? :laughing:

**Review by garethellis0 — DISMISSED** [review]
One comment, but other then that LGTM! :+1: :tada: - very nice, very clean

File: src/thunderbots/software/test/ai/game_state_play_selection.cpp (hunk 1 above)
Comment (jordanzliu) [reply src/thunderbots/software/test/ai/game_state_play_selection.cpp]: fixed

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Review by henryymliu — DISMISSED** [review]

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Nice work! This is cleaner than I was expecting.

My only request is that can we narrow the scope even further here? I'd like this test to be specifically for the `std::unique_ptr<Play> calculateNewPlay(const World &world);` in `stp.h`.

There are several reasons for this:
* This is _the_ function that does Play selection
* We want to test specifically this function, not anything that wraps it because the wrappers will return default plays and the test will always pass. This test should just cover the logic trying to choose a new valid play
* By scoping more specifically we don't need the extra "NO NAME" play stuff

This means that the test won't actually pass right now, so it's fine to comment it out to disable it when this PR is actually submitted. We will reactivate it in a week or so once we think we have full Play coverage.

**Comment by jordanzliu** [comment]
I'll make a test with narrowed scope, but I'll keep this one around as more of an integration test.

**Review by MathewMacDougall — DISMISSED** [review]
LGTM pending CI

**Review by MathewMacDougall — APPROVED** [review]
LGTM

**Pull request merged**